src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -56,10 +57,22 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(privateAssetsExclusionList);
             }
 
-            if(_filterlockFileTarget != null)
+            if(_packagesToBeFiltered != null)
             {
-                IEnumerable<LockFileTargetLibrary> filterLibraries = _filterlockFileTarget.Libraries;
-                Dictionary<string, LockFileTargetLibrary> filterLookup = filterLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+                var filterLookup = new Dictionary<string, HashSet<PackageInfo>>();
+                foreach (var pkg in _packagesToBeFiltered)
+                {
+                    if (filterLookup.ContainsKey(pkg.Name))
+                    {
+                        filterLookup[pkg.Name].Add(pkg);
```

Comment:
What happens when the entry XML file contains two entries with different versions but same cased name of the package?